server/src/search/search.controller.ts

```diff
@@ -21,27 +20,28 @@ const fullCourseSearch = async ({ search }: SearchQueryType) => {
   const query = search.getQuery();
   let fullSearch = new Set(); // set to ensure no duplicate courses
 
-  if (query !== undefined && query !== '') {
+  // checks query after at least 2 characters to speed up search
```

Comment:
I think this is a smart solution, since there isn't much information to search for with just one character. It can help improve speed without actually affecting the user experience